applications: Matter Bridge: extend CLI with data provider setters (#13282)
nrfconnect/sdk-nrf: applications/matter_bridge/README.rst

      uart:~$ matter_bridge add 256 "Kitchen Light"

Controlling a simulated OnOff Light bridged device
   Use the following command:

   .. parsed-literal::
      :class: highlight

      matter_bridge onoff *<new_state>* *<endpoint>*

   In this command:

   * *<new_state>*  is the new state (``0`` - off and ``1`` - on) that will be set on the simulated OnOff Light device.
   * *<endpoint>*  is the endpoint on which the bridged OnOff Light device is implemented

   Example command:

   .. code-block:: console

      uart:~$ matter_bridge onoff 1 3

   Note that the above command will only work if the :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` option is selected in the build configuration.

>    If the Kconfig option is not selected, the simulated device changes its state periodically in autonomous manner and can not be controlled by shell command.

Reviewer: ```suggestion
   If the Kconfig option is not selected, the simulated device changes its state periodically in autonomous manner and can not be controlled by using shell commands.
```

      uart:~$ matter_bridge add 256 "Kitchen Light"

Controlling a simulated bridged device
   Use the following command:

   .. parsed-literal::
      :class: highlight

      matter_bridge *<"bridged_device_label">* *<new_value>* *<endpoint>*

   In this command:

   * *<"bridged_device_label">* is the label of the simulated Matter bridged device type.
     The argument is mandatory and accepts the following values:

      * ``onoff`` for On/Off Light.
      * ``temperature`` for Temperature Sensor.
      * ``humidity`` for Humidity Sensor.

>    * *<new_value>*  is the new integer value that will be generated by the simulated bridged device (note that for ``onoff`` label only *0* and *1* are sensible values).

Reviewer 2: can you add information that 0 means off and 1 on?

Author: Yes, will add it.

>    * *<new_value>*  is the new integer value that will be generated by the simulated bridged device (note that for ``onoff`` label only *0* and *1* are sensible values).

Reviewer: ```suggestion
   * *<new_value>*  is the new integer value that will be generated by the simulated bridged device.
     For the ``onoff`` label, only ``0`` (off) and ``1`` (on) are sensible values.
```

   * *<endpoint>*  is the endpoint on which the bridged device is implemented

   Example commands:

   .. code-block:: console

      uart:~$ matter_bridge onoff 1 3

   .. code-block:: console

      uart:~$ matter_bridge temperature 25 4

   .. code-block:: console

      uart:~$ matter_bridge humidity 50 5

>    Note that above commands will only work if any of the following Kconfig options are selected in the build configuration:

Reviewer: ```suggestion
   Note that the above commands will only work if the Kconfig option corresponding to the simulated device is selected in the build configuration:
```

>    * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL`
>    * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_SHELL`

Reviewer: ```suggestion

   * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` for On/Off Light.
   * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_SHELL` for Temperature Sensor or Humidity Sensor.
```
Notice the empty line before the bullets.

>    Otherwise, the simulated devices generate data periodically in autonomous manner and can not be controlled by shell commands.

Reviewer: ```suggestion
   If the Kconfig option is not selected, the simulated devices generates data periodically in autonomous manner and can not be controlled by shell commands.
```

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED` - For the simulated bridged device.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_BT` - For the Bluetooth LE bridged device.

The simulated OnOff Light bridged device can operate in the following modes:

* Autonomous - The simulated device periodically changes its state.

> * Controllable - The user can explicitly control the On/Off state with shell command.

Reviewer: ```suggestion
* Controllable - The user can explicitly control the On/Off state by using shell commands.
```

> * Autonomous - The simulated device periodically changes its state.
> * Controllable - The user can explicitly control the On/Off state with shell command.
> 
> To build the simulated OnOff Light data provider in autonomous mode, select :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_AUTOMATIC` Kconfig option.
> To build the simulated OnOff Light data provider in controllable mode, select :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` Kconfig options.
> The :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` is enabled by default.

Reviewer: ```suggestion
* Autonomous - The simulated device periodically changes its state.
  To build the simulated OnOff Light data provider in this mode, select the :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_AUTOMATIC` Kconfig option.
* Controllable - The user can explicitly control the On/Off state with shell commands.
  To build the simulated OnOff Light data provider in this mode, select the :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` Kconfig option.
  This is enabled by default.
```
Since we're only dealing with one device type now, might as well consolidate the information.

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED` - For the simulated bridged device.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_BT` - For the Bluetooth LE bridged device.

> Moreover, the simulated bridged device can operate in the following modes:

Reviewer: ```suggestion
The simulated bridged device can operate in the following modes:
```

> * autonomous - it periodically generates new data
> * controllable - the user can explicitly control the generated data with shell commands

Reviewer: ```suggestion

* Autonomous - The simulated device periodically generates new data.
* Controllable - The user can explicitly control the generated data with shell commands.
```
Empty line again.

> To build the simulated data provider in autonomous mode, select:

Reviewer: ```suggestion
To build the simulated data provider in autonomous mode, select one of the following Kconfig options:
```

> * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_AUTOMATIC` for Humidity Sensor and Temperature Sensor data providers
> * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_AUTOMATIC` for On/Off Light data provider

Reviewer: ```suggestion

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_AUTOMATIC` - For Humidity Sensor and Temperature Sensor data providers.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_AUTOMATIC` - For On/Off Light data provider.
```
Empty line here as well.

> To build the simulated data provider in controllable mode, select:
> * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_SHELL` for Humidity Sensor and Temperature Sensor data providers
> * :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` for On/Off Light data provider

Reviewer: ```suggestion
To build the simulated data provider in controllable mode, select one of the following Kconfig options:

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_SENSOR_SHELL` - For Humidity Sensor and Temperature Sensor data providers.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED_ONOFF_SHELL` - For On/Off Light data provider.
```
